hack/lib/version.sh

```diff
@@ -50,6 +50,13 @@ kube::version::get_version_vars() {
 
     # Use git describe to find the version based on annotated tags.
     if [[ -n ${KUBE_GIT_VERSION-} ]] || KUBE_GIT_VERSION=$("${git[@]}" describe --tags --abbrev=14 "${KUBE_GIT_COMMIT}^{commit}" 2>/dev/null); then
+      # This translates the "git describe" to an actual semver.org
+      # compatible semantic version that looks something like this:
+      #   v1.1.0-alpha.0.6+84c76d1142ea4d
+      #
+      # TODO: We continue calling this "git version" because so many
+      # downstream consumers are expecting it there.
+      KUBE_GIT_VERSION=$(echo "${KUBE_GIT_VERSION}" | sed "s/-\([0-9]\{1,\}\)-g\([0-9a-f]\{14\}\)$/.\1\+\2/")
```

Comment:
Er, I'm not sure that's the right change to the regex, for the `master` case. (Sorry, back actually at a computer now.) My goal was something like `v1.1.0-alpha.0.6+84c76d1142ea4d`. The numeric `.6` past the `.0` actually force it to sort numerically, whereas semver says that `.0-6` sorts lexically, so when you get to `.0-10`, it's obnoxious (without doing, say, `printf %010d` or something`).